### PR TITLE
Cleanup & update dockerfile

### PR DESCRIPTION
**Pull request by aslafy-z**
This PR:

- Update docker `php` image
- Use curl to download prism
- Delete `prism_linux_amd64` from repo root folder
- Set `docker/entrypoint.sh` executable

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-php?pullRequest=741) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-php?pullRequest=741) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-php?pullRequest=741) it.</sub>

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,22 +1,18 @@
-FROM php:7.1-cli
+FROM php:7.2-cli
```
Comment (peter279k) [inline docker/Dockerfile]: Is this going to build the latest PHP version for the `Docker` environment?

Comment (aslafy-z) [reply docker/Dockerfile]: It's a prebuilt docker image, so yes it will use the said version.

**Comment by thinkingserious** [comment]
Hello @aslafy-z,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by childish-sambino** [comment]
Closing as prism integration has been moved to https://github.com/sendgrid/sendgrid-oai

**Pull request closed without merging**